boards/Makefile.common

```diff
@@ -63,15 +63,6 @@ ifeq ($(shell $(TOCK_ROOT_DIRECTORY)tools/semver.sh $(RUSTUP_VERSION) \< $(MINIM
   DUMMY := $(shell $(RUSTUP) update)
 endif
 endif
-
-# Verify that various required Rust components are installed. All of these steps
-# only have to be done once per Rust version, but will take some time when
-# compiling for the first time.
-ifneq ($(shell $(RUSTUP) target list | grep "$(TARGET) (installed)"),$(TARGET) (installed))
-  $(warning Request to compile for a missing TARGET, make will install in 5s)
-  $(warning Consider updating 'targets' in 'rust-toolchain.toml')
-  $(shell sleep 5s && $(RUSTUP) target add $(TARGET))
-endif
```

Comment:
To clarify, cargo does not do this automatically, `rustup` does it iff the target is included in `rust-toolschain.toml`. I'm not a fan of any of this setup being in the Makefile to begin with (we needed to add a hack to be able to get around this on NixOS et al where this totally doesn't work), but I don't see this as a related change to this PR.

Reply:
Fair enough. Bad commit message on my part. I'm happy to rename this PR "refactor build system and review makefile".

Reply:
It's not important, I was just noting as I read through.